src/github.com/matrix-org/dendrite/roomserver/storage/previous_events_table.go

```diff
@@ -0,0 +1,72 @@
+package storage
+
+import (
+	"database/sql"
+	"github.com/matrix-org/dendrite/roomserver/types"
+)
+
+const previousEventSchema = `
+-- The previous events table stores the event_ids referenced by the events
+-- stored in the events table.
+-- This is used to tell if a new event is already referenced by an event we 
+CREATE TABLE IF NOT EXISTS previous_events (
+	// The string event ID taken from the prev_events key of an event.
+	previous_event_id TEXT NOT NULL,
+	// The SHA256 reference hash taken from the prev_events key of an event.
+	previous_reference_sha256 BYTEA NOT NULL,
+	// A list of numeric event IDs of events that reference this prev_event.
+	// This isn'
+	event_nids BIGINT[] NOT NULL,
+	CONSTRAINT previous_event_id_unique UNIQUE (region_nid, prev_event_id, prev_reference_sha256)
```

Comment:
wut? `prev_event_id` isn't a thing, it's `previous_event_id`. Ditto with the SHA. What's a `region_nid`?